py-polars/polars/internals/expr.py

```diff
@@ -796,7 +803,7 @@ def shift(self, periods: int = 1) -> "Expr":
         """
         return wrap_expr(self._pyexpr.shift(periods))
 
-    def shift_and_fill(self, periods: int, fill_value: "Expr") -> "Expr":
+    def shift_and_fill(self, periods: int, fill_value: Union[int, "Expr"]) -> "Expr":
```

Comment:
I think that it may be any value that we can turn into an `Expr` literal, so `flt, bool, int, str`

Reply:
I was thinking the same, did here the minimum to make `mypy` happy. Should it be `Any` in that case, or do we want to limit to a subset (bool, int, float, str, date, ?)?

Reply:
I think limit to that subset, for other types we cannot create a `pl.lit`